arithmetization/src/main/java/net/consensys/linea/plugins/rpc/tracegeneration/GenerateConflatedTracesV2.java

```diff
@@ -38,16 +39,27 @@
  * based on the provided request parameters and writes them to a file.
  */
 @Slf4j
-@RequiredArgsConstructor
 public class GenerateConflatedTracesV2 {
   private static final JsonConverter CONVERTER = JsonConverter.builder().build();
   private static final String TRACE_FILE_EXTENSION = ".lt";
   private static final String TRACE_TEMP_FILE_EXTENSION = ".lt.tmp";
 
-  private final BesuContext besuContext;
+  private final RequestLimiter<PluginRpcRequest, TraceFile> requestLimiter;
+
   private final Path tracesOutputPath;
   private TraceService traceService;
 
+  public GenerateConflatedTracesV2(
+      final BesuContext besuContext, final TracesEndpointConfiguration endpointConfiguration) {
+    this.traceService = Requests.getTraceService(besuContext);
+    this.requestLimiter =
```

Comment:
this limiter shoul dbe the same as for the line counts.

Reply:
There is a slight technical difficulty to do that, so we need to discuss this aspect.